readjson.go

```diff
@@ -79,19 +79,17 @@ func readData(device string) (gjson.Result, error) {
 		timeToScan := false
 		if cacheOk {
 			timeToScan = time.Now().After(cacheValue.LastCollect.Add(options.SMARTctl.CollectPeriodDuration))
-		} else {
-			timeToScan = true
 		}
 
-		if timeToScan {
+		if timeToScan || !cacheOk {
```

Comment:
Perhaps better to combine checks - for `cacheOk` first and only then calculate `timeToScan`, otherwise `cacheOk` will be verified 2 times.
```
if !cacheOk || time.Now().After(cacheValue.LastCollect.Add(options.SMARTctl.CollectPeriodDuration)) {
...
}
```